Marlin/src/gcode/control/M42.cpp

```diff
@@ -55,13 +62,20 @@ void GcodeSuite::M42() {
 
   if (!parser.boolval('I') && pin_is_protected(pin)) return protected_pin_err();
 
+  bool avoidWrite = false;
   if (parser.seenval('M')) {
     switch (parser.value_byte()) {
-      case 0: pinMode(pin, INPUT); break;
+      case 0: pinMode(pin, INPUT); avoidWrite = true; break;
```

Comment:
maybe we could just put "return;" instead of break on the inputs... but will skip the error message...

Reply:
I thing the error message is a good thing.